x-pack/plugin/otel-data/src/yamlRestTest/resources/rest-api-spec/test/20_metic_tests.yml

```
@@ -0,0 +1,88 @@
+---
+setup:
+  - do:
+      cluster.health:
+        wait_for_events: languid
+  - do:
+      cluster.put_component_template:
+        name: metrics-otel@custom
+        body:
+          template:
+            settings:
+              index:
+                routing_path: [unit, attributes.*, resource.attributes.*]
+                mode: time_series
+                time_series:
+                  start_time: 2024-07-01T13:03:08.138Z
+---
+"Test push service overview metric":
+  - do:
+      indices.get_index_template:
+        name: metrics-otel@template
+  - length: {index_templates: 1}
+  - do:
+      bulk:
+        index: metrics-generic.otel-default
+        refresh: true
+        body:
+          - create: {}
+          - '{"@timestamp":"2024-07-18T14:48:33.467654000Z","data_stream":{"dataset":"generic.otel","namespace":"default","type":"metrics"},"processor":{"event":"metric"},"resource":{"attributes":{"service.name":"OtelSample","telemetry.sdk.language":"dotnet","telemetry.sdk.name":"opentelemetry"},"dropped_attributes_count":0}}'
+  - is_false: errors
+  - do:
+      search:
+        index: metrics-generic.otel-default
+        body:
+          fields: ["service.name", "telemetry.sdk.language", "telemetry.sdk.name" ]
+  - length: { hits.hits: 1 }
+  - match: { hits.hits.0.fields: {"service.name": ["OtelSample"], "telemetry.sdk.language": ["dotnet"], "telemetry.sdk.name": ["opentelemetry"] }}
```

I'd prefer this in terms of readability and consistency with other tests in the codebase
```suggestion
  - match: { hits.hits.0.fields.service\.name:  ["OtelSample"] }  
  - match: { hits.hits.0.fields.telemetry\.sdk\.language:  ["dotnet"] }
  ...
```

Also applies to other places in this and other test files where assertions on `fields` are made.

---

Pushed it in https://github.com/elastic/elasticsearch/pull/111091/commits/d5de2a1625866f227c84efd3c44b198dc2659a65.